### DISCUSSION TITLE
Beginner question - How do i compile this java project?

### DISCUSSION THREAD

**Discussion by Zhao-shenghao** [github/securitylab#163] [General] [answered]
![89011875-c1728980-d343-11ea-866e-8af8e554bbb1](https://user-images.githubusercontent.com/43957169/89093147-82e3da00-d3ea-11ea-98a6-c14cd25f7b59.png)

In fact, I also encountered this problem when compiling other projects，such as archiva,commons-fileupload,hadoop...

**Comment by jbj** [comment, marked as answer]
I probably can't help you either way, but I expect that other people can help you better if you obtain an English version of the error message.

Depending on what you're trying to do, you may not need to compile these projects. There are pre-compiled databases for many popular projects available on lgtm.com. A Tomcat Java snapshot is available here: https://lgtm.com/projects/g/apache/tomcat/ci/#ql. You can also see security alerts and run your own queries directly on lgtm.com.

**Reply by Zhao-shenghao** [reply]
First of all thank you for replying to help me.
In fact, this picture is the English version. The sixth and seventh lines are some random Chinese code, I also don’t know what they mean.Maybe they are not important.
[https://lgtm.com/projects/g/apache/tomcat/ci/#ql](url) is indeed a compiled database, but it seems that he only provides the latest version. But my research needs to use some historical versions, and I don't know how to go back there.

**Reply by jbj** [reply]
I'm afraid there's no way to obtain historical databases on LGTM, so you're on the right track trying to build them yourself.

The messages from CodeQL before and after the Chinese text don't have much information in them. The "No source code was seen" message just means that the final step of the database creation can't run because the build didn't run.

Have you tried building the project without the involvement of CodeQL? Does it build if you type the command that's echoed in your screenshot:

```
ant -noinput -f "build.xml" clean "deploy"
```

I checked that this is the same command used on https://lgtm.com/projects/g/apache/tomcat/logs/languages/lang:java, so it ought to be right for this project.
